pylearn2/sandbox/nlp/models/mlp.py

```diff
@@ -189,3 +198,303 @@ def get_l1_weight_decay(self, coeff):
         assert isinstance(coeff, float) or hasattr(coeff, 'dtype')
         W, = self.transformer.get_params()
         return coeff * abs(W).sum()
+
+
```

Comment:
About negative sampling, maybe this helps: [http://arxiv.org/abs/1402.3722](http://arxiv.org/abs/1402.3722).